src/components/Posts/PostCollect.tsx

```diff
@@ -0,0 +1,373 @@
+
+        if (post.hasActed) {
+            return (
+                <>
+                    <Trans>Collected</Trans>
+                    {contractExploreUrl ? (
+                        <Link
+                            className="ml-1"
+                            href={EVMExplorerResolver.addressLink(polygon.id, contractExploreUrl) ?? ''}
+                            target="_blank"
+                            rel="noreferrer noopener"
+                        >
+                            <LinkIcon width={18} height={18} />
+                        </Link>
+                    ) : null}
+                </>
+            );
+        }
+
+        if (isSoldOut)
+            return (
+                <>
+                    <Trans>Sold Out</Trans>
+                    {contractExploreUrl ? (
+                        <Link
+                            className="ml-1"
+                            href={EVMExplorerResolver.addressLink(polygon.id, contractExploreUrl) ?? ''}
+                            target="_blank"
+                            rel="noreferrer noopener"
+                        >
+                            <LinkIcon width={18} height={18} />
+                        </Link>
+                    ) : null}
+                </>
+            );
+
+        if (isTimeOut)
+            return (
+                <>
+                    <Trans>Time Out</Trans>
+                    {contractExploreUrl ? (
+                        <Link
+                            href={EVMExplorerResolver.addressLink(polygon.id, contractExploreUrl) ?? ''}
+                            target="_blank"
+                            rel="noreferrer noopener"
+                        >
+                            <LinkIcon width={18} height={18} />
+                        </Link>
+                    ) : null}
+                </>
+            );
+
+        if (!allowed) {
+            return <Trans>Allow Collect Module</Trans>;
+        }
+
+        if (!hasEnoughBalance) {
+            return <Trans>Insufficient Balance</Trans>;
+        }
+
+        if (post.collectModule?.amount && post.collectModule.currency) {
+            return (
+                <Trans>
+                    Collect for {post.collectModule.amount} {post.collectModule.currency}
+                </Trans>
+            );
+        }
+
+        if (post.collectModule?.followerOnly && !profile?.viewerContext?.following) {
+            return <Trans>Follow to Collect</Trans>;
+        }
+
+        return <Trans>Free Collect</Trans>;
+    }, [
+        post.collectModule?.contract.address,
+        post.collectModule?.amount,
+        post.collectModule?.currency,
+        post.collectModule?.followerOnly,
+        post.hasActed,
+        isLogin,
+        isSoldOut,
+        isTimeOut,
+        allowed,
+        hasEnoughBalance,
+        profile?.viewerContext?.following,
+    ]);
+
+    const handleClick = useCallback(async () => {
```

Comment:
Handling the loading of async function. Present a loading indicator.

```suggestion
    const handleClick = useAsyncFn(async () => {
```